h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstGroup.java

```diff
@@ -673,24 +688,20 @@ public MedianResult(int len) {
   // long) that defines the Group.  Also contains an array of doubles for the
   // aggregate results, one per aggregate.
   public static class G extends Iced {
-    public final double[] _gs;  // Group Key: Array is final; contents change with the "fill"
+//    public final double[] _gs;  // Group Key: Array is final; contents change with the "fill"
+    public final byte[][] _gsB;
     int _hash;           // Hash is not final; changes with the "fill"
 
     public final double _dss[][];      // Aggregates: usually sum or sum*2
     public final long _ns[];         // row counts per aggregate, varies by NA handling and column
-/*    int[] _medianCols;    // record which columns in reference to data frame
-    double[] _medians;
-    boolean[] _isMedian;
-    int[] _newChunkCols;  // record which columns in newChunk to store group
-    public NAHandling[] _na;*/
     public MedianResult medianR = null;
 
     public G(int ncols, AGG[] aggs) {
       this(ncols, aggs, false);
     }
 
     public G(int ncols, AGG[] aggs, boolean hasMedian) {
-      _gs = new double[ncols];
+      _gsB = new byte[ncols][];
```

Comment:
I thought we would use just a single byte array to store everything -this way you would get the most efficient compressed representation

the layout is known - we do know how many bytes we need to represent each value

Reply:
I thought about it but the problem is that we actually use the actual value of each column to perform calculations in later parts.  If you combine all the keys into one byte array, I will need another array to remember which column bytes starts and which one ends....

Reply:
yes, however you need this array just once - all rows can share the same layout